tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -7536,6 +7536,90 @@ public function dataArrayPointerFunctions(): array
 				'\'bar\'|\'baz\'',
 				'end($secondConditionalArray)',
 			],
+			[
+				'mixed',
+				'current()',
+			],
+			[
+				'stdClass|false',
+				'current($generalArray)',
+			],
+			[
+				'mixed',
+				'current($somethingElse)',
+			],
+			[
+				'false',
+				'current($emptyConstantArray)',
+			],
+			[
+				'1|2|false',
```

Comment:
@ondrejmirtes now `current`, `prev` and `next` always report all possible return types, but I think this PR will be useless anyway:
```php
class A { function foo() {} }

/** @var \A[] $aa */
$aa = [new \A()];

$a = current($aa);
$a->foo();
```
now reports:
```
 ------ -------------------------------------- 
  Line   tmp.php                                 
 ------ -------------------------------------- 
   9     Cannot call method foo() on A|false.  
 ------ -------------------------------------- 
```
For sure adding something like `\Webmozart\Assert\Assert::isInstanceOf($a, \A::class);` before the `current` clears away the error, but nullifies the point of this PR of not having to add more code or more `ignoreErrors` :disappointed:

Reply:
For sake of completeness, even adding a `\Webmozart\Assert\Assert::notEmpty($aa)` before the `current` doesn't clear the error away, neither will do the same `static::assertNotEmpty($aa)` in tests context :disappointed: